include/seqan3/utility/views/zip.hpp

```diff
@@ -6,26 +6,476 @@
+            return sentinel<false>(zip::tuple_transform(std::ranges::end, views_));
+        else if constexpr ((std::ranges::random_access_range<Views> && ...))
+            return begin() + std::iter_difference_t<iterator<false>>(size());
+        else
+            return iterator<false>(zip::tuple_transform(std::ranges::end, views_));
+    }
+
+    constexpr auto end() const
+        requires (std::ranges::range<Views const> && ...)
+    {
+        if constexpr (!zip::zip_is_common<Views const...>)
+            return sentinel<true>(zip::tuple_transform(std::ranges::end, views_));
+        else if constexpr ((std::ranges::random_access_range<Views const> && ...))
+            return begin() + std::iter_difference_t<iterator<true>>(size());
+        else
+            return iterator<true>(zip::tuple_transform(std::ranges::end, views_));
+    }
+
+    constexpr auto size()
+        requires (std::ranges::sized_range<Views> && ...)
+    {
+        return std::apply([] (auto... sizes)
+            {
+                using common_size_t = std::make_unsigned_t<std::common_type_t<decltype(sizes)...>>;
+                return std::ranges::min({static_cast<common_size_t>(sizes)...});
+            },
+            zip::tuple_transform(std::ranges::size, views_));
+    }
+
+    constexpr auto size() const
+        requires (std::ranges::sized_range<Views const> && ...)
+    {
+        return std::apply([] (auto... sizes)
+            {
+                using common_size_t = std::make_unsigned_t<std::common_type_t<decltype(sizes)...>>;
+                return std::ranges::min({static_cast<common_size_t>(sizes)...});
+            },
+            zip::tuple_transform(std::ranges::size, views_));
+    }
+};
+
+// Here be dragons: May need handling of rvalues (pre gcc12). Seems to work.
+template <typename... range_ts>
+zip_view(range_ts && ...) -> zip_view<std::ranges::views::all_t<range_ts>...>;
+
+namespace zip
+{
+
+#if defined(__GNUC__) && (__GNUC__ == 10) // cpp17 iterators
+template <bool is_const, typename... range_ts>
+concept all_contiguous =
+    (std::ranges::contiguous_range<view_helper::maybe_const<is_const, range_ts>> && ...);
+
+template <bool Const, typename... Views>
+struct iterator_category_t
+{
+    using iterator_category =
+        std::conditional_t<all_contiguous<Const, Views...>, std::contiguous_iterator_tag,
+            std::conditional_t<all_random_access<Const, Views...>, std::random_access_iterator_tag,
+                std::conditional_t<all_bidirectional<Const, Views...>, std::bidirectional_iterator_tag,
+                    std::conditional_t<all_forward<Const, Views...>, std::forward_iterator_tag,
+                        std::input_iterator_tag>>>>;
+};
+#else // cpp20 iterators
+template <bool> struct iterator_category_t;
+template <> struct iterator_category_t<true> { using iterator_category = std::input_iterator_tag; };
+template <> struct iterator_category_t<false> {};
+#endif
+
+} // namespace seqan3::detail::zip
```

Comment:
Same here, shouldn't this just say (not sure what our coding guideline says)
```suggestion
} // namespace zip
```